core/spam/engine.go

```diff
@@ -231,3 +235,60 @@ func (e *Engine) PostBlockAccept(tx abci.Tx) (bool, error) {
 	}
 	return e.transactionTypeToPolicy[command].PostBlockAccept(tx)
 }
+
+func parseBannedUntil(until int64) *string {
+	if until == 0 {
+		return nil
+	}
+	t := strconv.FormatInt(until, 10)
+	return &t
+}
+
+func (e *Engine) GetSpamStatistics(partyID string) *protoapi.SpamStatistics {
+	stats := &protoapi.SpamStatistics{}
+
+	for txType, policy := range e.transactionTypeToPolicy {
+		pStats := policy.GetStats(partyID)
+
+		ssProto := &protoapi.SpamStatistic{}
+		var bannedUntil int64
+
+		if len(pStats) > 0 {
+			ssProto = &protoapi.SpamStatistic{
+				CountForEpoch: pStats[0].Total,
+				MaxForEpoch:   pStats[0].Limit,
+				BannedUntil:   parseBannedUntil(pStats[0].BannedUntil),
+			}
+
+			bannedUntil = pStats[0].BannedUntil
+		}
+
+		switch txType {
+		case txn.ProposeCommand:
+			stats.Proposals = ssProto
+		case txn.DelegateCommand:
+			stats.Delegations = ssProto
+		case txn.TransferFundsCommand:
+			stats.Transfers = ssProto
+		case txn.AnnounceNodeCommand:
+			stats.NodeAnnouncements = ssProto
+		case txn.VoteCommand:
+			vs := make([]*protoapi.VoteSpamStatistic, len(pStats))
+			for i, ps := range pStats {
+				vs[i] = &protoapi.VoteSpamStatistic{
```

Comment:
it's a bit redundant to generate the struct there with the redundant banned until and then copy it here. better to return already the full struct from the vote stats func.